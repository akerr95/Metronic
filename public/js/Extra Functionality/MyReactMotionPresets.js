/**
 * Created by akerr on 9/21/16.
 */
import {spring} from "react-motion"
export const ICON = {
    scale: 1,
    factor:0.35,
    extra: {
        stiffness: 200,
        damping: 10
    }
};
export const NOTICE = {
    scale: 1,
    factor:0.2,
    extra: {
        stiffness: 110,
        damping: 5
    }
};
export const USERPROFILE = {
    scale: 1,
    factor:0.2,
    extra: {
        stiffness: 110,
        damping: 5
    }
};
export const presets= {
    genScale: function (subject,element) {
        return {scale: spring(this.clicked(subject.iconName,subject.state)?element.scale+element.factor:element.scale,element.extra)};
    },
    genOpacity:function(subject,element){
        return {opacity: spring(this.clicked(subject.iconName,subject.state)?element.opacity+element.factor:element.opacity,element.extra)};
    },
    clicked: function (iconName,state){
        let isClicked=false;
        if(state.has(iconName)){
            isClicked =state.get(iconName);
        }
        return isClicked;
    }
};

