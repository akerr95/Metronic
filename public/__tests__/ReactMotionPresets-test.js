/**
 * Created by akerr on 9/21/16.
 */
import {presets,ICON} from '../js/Extra Functionality/MyReactMotionPresets.js';


describe("Presets",()=>{
    it("Presets should be defined",()=>{
        let result = presets;
        expect(result).not.toBeUndefined();
        expect(result).toMatchSnapshot();
    });
    it("Presets genScale should return an object with scale",()=>{
        let iconName = "test-1";
        let state = new Map([[iconName,false]]);
        let result = presets.genScale({iconName,state},ICON);

        expect(result.hasOwnProperty("scale")).toBeTruthy();
        expect(result.scale).toMatchSnapshot();
    });
    it("Presets genOpacity should return an object with opacity",()=>{
        let iconName = "test-1";
        let state = new Map([[iconName,false]]);
        let result = presets.genOpacity({iconName,state},ICON);

        expect(result.hasOwnProperty("opacity")).toBeTruthy();
        expect(result.opacity).toMatchSnapshot();
    });

});
