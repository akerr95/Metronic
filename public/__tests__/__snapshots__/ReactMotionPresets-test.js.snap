exports[`Presets Presets genOpacity should return an object with opacity 1`] = `
Object {
  "damping": 40,
  "precision": 0.01,
  "stiffness": 300,
  "val": undefined
}
`;

exports[`Presets Presets genScale should return an object with scale 1`] = `
Object {
  "damping": 40,
  "precision": 0.01,
  "stiffness": 300,
  "val": 1
}
`;

exports[`Presets Presets should be defined 1`] = `
Object {
  "clicked": [Function clicked],
  "genOpacity": [Function genOpacity],
  "genScale": [Function genScale]
}
`;
